File: four_pillars_of_oop/no_01_encapsulation/no_03_modularity/java/Employee.java
package four_pillars_of_oop.no_01_encapsulation.no_03_modularity.java;

public class Employee {
    private String name;  // Private attributes
    private double salary;

    public Employee(String name, double salary) {
        this.name = name;
        setSalary(salary);
    }

    public void setSalary(double salary) {
        if (salary > 0) {
            this.salary = salary;
        } else {
            System.out.println("Salary must be positive.");
        }
    }

    public double getSalary() {
        return salary;
    }

    public String getDetails() {
        return "Employee: " + name + ", Salary: $" + salary;
    }
}

File: four_pillars_of_oop/no_01_encapsulation/no_03_modularity/python/employee.py
class Employee:
    def __init__(self, name, salary):
        self.__name = name
        self.__salary = salary

    def set_salary(self, salary):
        if salary > 0:
            self.__salary = salary
        else:
            print("Salary must be positive.")

    def get_salary(self):
        return self.__salary

    def get_details(self):
        return f"Employee: {self.__name}, Salary: ${self.__salary}"
